### PR TITLE
Fix backdoor API provided on health endpoint

### LINKED ISSUE #5906: bug: backdoor API provided on health endpoint does not work
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

The backdoor API to restart the process provided on `/health` endpoint does not work. 

```shell
❯ docker run -p 4566:4566 -d localstack/localstack:latest
❯ curl -XPOST http://127.0.0.1:4566/health -H "Content-Type: application/json" -d  '{ "action": "restart" }'
❯ docker logs -f 92e3c466b450
Waiting for all LocalStack services to be ready
2022-04-21 12:23:11,211 CRIT Supervisor is running as root.  Privileges were not dropped because no user is specified in the config file.  If you intend to run as root, you can set user=root in the config file to avoid this message.
2022-04-21 12:23:11,215 INFO supervisord started with pid 15
2022-04-21 12:23:12,226 INFO spawned: 'infra' with pid 21

LocalStack version: 0.14.2
LocalStack build date: 2022-04-20
LocalStack build git hash: f9aa2f55

2022-04-21 12:23:13,419 INFO success: infra entered RUNNING state, process has stayed up for > than 1 seconds (startsecs)
Starting edge router (https port 4566)...
Ready.
[2022-04-21 12:23:13 +0000] [21] [INFO] Running on https://0.0.0.0:4566 (CTRL + C to quit)
2022-04-21T12:23:13.529:INFO:hypercorn.error: Running on https://0.0.0.0:4566 (CTRL + C to quit)
2022-04-21T12:23:27.032:WARNING:localstack.utils.server.http2_server: Error in proxy handler for request POST http://127.0.0.1:4566/health: 'dict' object is not callable Traceback (most recent call last):
  File "/opt/code/localstack/localstack/utils/server/http2_server.py", line 185, in index
    result = await run_sync(handler, request, data)
  File "/opt/code/localstack/localstack/utils/asyncio.py", line 85, in run_sync
    return await loop.run_in_executor(thread_pool, copy_context().run, func_wrapped)
  File "/opt/code/localstack/localstack/utils/threads.py", line 38, in run
    result = self.func(self.params, **kwargs)
  File "/opt/code/localstack/localstack/utils/asyncio.py", line 30, in _run
    return fn(*args, **kwargs)
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 972, in handler
    return modify_and_forward(
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 516, in wrapper
    value = func(*args, **kwargs)
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 596, in modify_and_forward
    listener_result = listener.forward_request(
  File "/opt/code/localstack/localstack/services/internal.py", line 213, in forward_request
    return super().forward_request(method, path, data, headers)
  File "/opt/code/localstack/localstack/http/adapters.py", line 63, in forward_request
    return super().forward_request(method, path, data, headers)
  File "/opt/code/localstack/localstack/http/adapters.py", line 38, in forward_request
    response = self.request(request)
  File "/opt/code/localstack/localstack/http/adapters.py", line 71, in request
    return self.router.dispatch(request)
  File "/opt/code/localstack/localstack/http/router.py", line 149, in dispatch
    return self.dispatcher(request, handler, args)
  File "/opt/code/localstack/localstack/http/dispatcher.py", line 63, in _dispatch
    result = fn(request, **args)
  File "/opt/code/localstack/localstack/services/internal.py", line 39, in on_post
    data = request.json()
TypeError: 'dict' object is not callable

2022-04-21T12:23:27.033:INFO:localstack.utils.threads: Thread run method <function AdaptiveThreadPool.submit.<locals>._run at 0x7f9b4ff5e9d0>(None) failed: 'dict' object is not callable Traceback (most recent call last):
  File "/opt/code/localstack/localstack/utils/threads.py", line 38, in run
    result = self.func(self.params, **kwargs)
  File "/opt/code/localstack/localstack/utils/asyncio.py", line 30, in _run
    return fn(*args, **kwargs)
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 972, in handler
    return modify_and_forward(
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 516, in wrapper
    value = func(*args, **kwargs)
  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 596, in modify_and_forward
    listener_result = listener.forward_request(
  File "/opt/code/localstack/localstack/services/internal.py", line 213, in forward_request
    return super().forward_request(method, path, data, headers)
  File "/opt/code/localstack/localstack/http/adapters.py", line 63, in forward_request
    return super().forward_request(method, path, data, headers)
  File "/opt/code/localstack/localstack/http/adapters.py", line 38, in forward_request
    response = self.request(request)
  File "/opt/code/localstack/localstack/http/adapters.py", line 71, in request
    return self.router.dispatch(request)
  File "/opt/code/localstack/localstack/http/router.py", line 149, in dispatch
    return self.dispatcher(request, handler, args)
  File "/opt/code/localstack/localstack/http/dispatcher.py", line 63, in _dispatch
    result = fn(request, **args)
  File "/opt/code/localstack/localstack/services/internal.py", line 39, in on_post
    data = request.json()
TypeError: 'dict' object is not callable
```

### Expected Behavior

The backdoor API restarts the process successfully.

### How are you starting LocalStack?

With a `docker run` command

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

* `docker run -p 4566:4566 -d localstack/localstack:latest`

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

* `curl -XPOST http://127.0.0.1:4566/health -H "Content-Type: application/json" -d  '{ "action": "restart" }'`

### Environment

```markdown
- OS: macOS v12.2 (Monterey)
- LocalStack: latest
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by nao23**
fixes #5906 

Currently, backdoor API to restart the process provided on `/health` endpoint does not work.
This patch fixes some errors to work correctly.

### CODE REVIEW

File: localstack/services/internal.py [hunk 1]
```diff
@@ -36,7 +36,7 @@ def __init__(self, service_manager) -> None:
         self.state = {}
 
     def on_post(self, request):
-        data = request.json()
```
Comment (nao23) [inline localstack/services/internal.py:39]: It causes a type error:
```
TypeError: 'dict' object is not callable
```

File: localstack/services/infra.py [hunk 2]
```diff
@@ -338,7 +338,7 @@ def terminate_all_processes_in_docker():
     print("INFO: Received command to restart all processes ...")
     cmd = (
         'ps aux | grep -v supervisor | grep -v docker-entrypoint.sh | grep -v "bin/localstack" | '
-        "grep -v localstack_infra.log | awk '{print $1}' | grep -v PID"
+        "grep -v localstack_infra.log | awk '{print $2}' | grep -v PID"
```
Comment (nao23) [inline localstack/services/infra.py:341]: Fix awk command to get pids correctly from `ps aux` result.

File: localstack/services/infra.py [hunk 3]
```diff
@@ -352,7 +352,7 @@ def terminate_all_processes_in_docker():
             except Exception:
                 pass
     # kill the process itself
-    sys.exit(0)
```
Comment (nao23) [inline localstack/services/infra.py:355]: `sys.exit(0)` only exit current thread so that process itself is not exit.

**Comment by thrau** [comment]
thanks for uncovering this @nao23, since there are a couple of other issues related to this, i'd like to push directly into the PR if that's OK.

File: localstack/services/infra.py (hunk 2 above)
Comment (thrau) [reply localstack/services/infra.py:341]: great catch

File: localstack/services/infra.py (hunk 3 above)
Comment (thrau) [reply localstack/services/infra.py:355]: it only terminates the interpreter if `sys.exit` is called from the main thread

**Review by thrau — COMMENTED** [review]
thanks @nao23, going to push a couple of additional fixes and then run the CI suite

**Comment by nao23** [comment]
@thrau thank you for your review!

**Pull request merged**